### ISSUE TITLE
Error while trying to use a different adapter (ie. curl-client instead of guzzle6-adapter)

### ISSUE THREAD

**Issue by gpou** [Payum/Payum#591]
So, I'm trying to use Payum along with php-http/curl-client instead of php-http/guzzle6-adapter.
I need to do so because I am using an older version of Guzzle, and right now I cannot upgrade.

I am calling the PayumBuilder like this:

```
        $this->payum = (new PayumBuilder())
            ->addDefaultStorages()
            ->addGateway($payumGatewayFactoryName, [
                    ...
                    'httplug.message_factory'=> $myMessageFactory,
                    'httplug.stream_factory' => $myStreamFactory,
                    'httplug.client' => new HttpCurlClient($myMessageFactory, $myStreamFactory),
            ]);
```

An error is thrown: "This version of Guzzle is not supported." Coming from here: https://github.com/Payum/Payum/blob/master/src/Payum/Core/Bridge/Guzzle/HttpClientFactory.php#L29

And the error is due to this deprecated setting here: https://github.com/Payum/Payum/blob/master/src/Payum/Core/CoreGatewayFactory.php#L136

```
            // deprecated
            'guzzle.client' => HttpClientFactory::createGuzzle(),
```

So, if this guzzle.client is deprecated and not used anywhere, can it be removed please?

**Comment by makasim** [comment]
I'll take a look

**Comment by gpou** [comment]
@makasim Thanks a lot

**Comment by gpou** [comment]
@makasim Done: https://github.com/Payum/Payum/pull/593

I didn't add anything to UPGRADE.md because it was already there:
https://github.com/gpou/Payum/blob/master/UPGRADE.md#130
> [gateway-factory] Option 'guzzle.client' was removed.

**Comment by makasim** [comment]
thanks for the fix,I tagged a new minor version: 1.3.8

**Comment by gpou** [comment]
Thank you @makasim !
